lib/srv/app/session.go

```diff
@@ -192,34 +214,57 @@ func (s *Server) newStreamer(ctx context.Context, sessionID string, recConfig ty
 	return fileStreamer, nil
 }
 
+// createTracker creates a new session tracker for the app session.
+func (s *Server) createTracker(sess *session, identity *tlsca.Identity) error {
+	s.log.Debugf("Creating tracker for session %v", sess.id)
+	initiator := &types.Participant{
+		ID:   identity.Username,
+		User: identity.Username,
+	}
+
+	tracker, err := types.NewSessionTracker(types.SessionTrackerSpecV1{
+		SessionID:    sess.id,
+		Kind:         string(types.AppSessionKind),
+		State:        types.SessionState_SessionStateRunning,
```

Comment:
Should State be set by CreateSessionTracker? It seems strange to impose an state from outside.

Reply:
What do you mean exactly? The tracker state doesn't impose anything here, it's just meant to keep track of the session state. Do you think it should start at `pending` and be updated later in the session creation cycle?

Reply:
I was thinking the innermost CreateSessionTracker could transition it.

Reply:
Ah, do you mean that it should be set on the receiving side of the `CreateSessionTracker` request? If so I disagree. The session lives on the client side of the request, and updates are sent to the server side for bookkeeping.

For a more specific example, ssh and kube session both create session trackers in the `pending` state, and update them to `running` after going through the moderated sessions check. So it is a requirement for the client to set the state here.

Sorry if I'm still misunderstanding.

Reply:
That makes sense, thanks for explaining.